### PR TITLE
Add mapview ornaments layout example

### PR DESCRIPTION
**Pull request by lloydsheng**
Now that 4.10.0 released, We need to add an example for https://github.com/mapbox/mapbox-gl-native/pull/13911

### CODE REVIEW

**Comment by friedbunny** [comment]
@lloydsheng @jmkiley anything blocking this?

**Comment by jmkiley** [comment]
👋 @lloydsheng do you need any help finishing this example?

**Comment by julianrex** [comment]
@lloydsheng can @m-stephen help get this in, so we can start seeing it with the 5.4.0 release?

**Comment by lloydsheng** [comment]
@jmkiley  Thanks for your review. I've fixed the issues and added some comments. Please see if there are any other issues.

/cc @julianrex

File: Examples/Examples.m [hunk 1]
```diff
@@ -65,7 +65,8 @@ + (NSArray *)groups {
                     @{@"className": MBXExampleFeatureSelection, @"title": @"Select a feature within a layer"},
                     @{@"className": MBXExampleSelectFeature, @"title": @"Select a water feature"},
                     @{@"className": MBXExampleUserTrackingModes, @"title": @"Switch between user tracking modes"},
-            ]
+                    @{@"className": MBXExampleOrnamentsLayout, @"title": @"Ornaments layout"}
+                    ]
```
Comment (friedbunny) [inline Examples/Examples.m]:
```suggestion
            ]
```

File: Examples/Swift/OrnamentsLayoutExample.swift [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    var currentPositionIndex: Int = 0
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        mapView = MGLMapView(frame: view.bounds)
+        mapView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        mapView.showsScale = true
+        mapView.setCenter(CLLocationCoordinate2DMake(39.915143, 116.404053), zoomLevel: 16, direction: 30, animated: false)
+        self.view.addSubview(mapView)
```
Comment (friedbunny) [inline Examples/Swift/OrnamentsLayoutExample.swift]: Nit: omit `self` everywhere in this Swift implementation (unless forced to use it by being inside a block, etc.).

File: Examples/ObjectiveC/OrnamentsLayoutExample.m [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    self.timer = [NSTimer scheduledTimerWithTimeInterval:1
+                                                  target:self
+                                                selector:@selector(onTimerTick)
+                                                userInfo:nil
+                                                 repeats:YES];
+}
+
+- (void)updateOrnamentsPosition {
+    // Get position matrix for current turn. We mod 4 to roll over the matrix array.
+    MGLOrnamentPosition *currentPosition = ornamentPositions[_currentPositionIndex % 4];
```
Comment (friedbunny) [inline Examples/ObjectiveC/OrnamentsLayoutExample.m]: Nit: use `self` consistently.

```suggestion
    MGLOrnamentPosition *currentPosition = ornamentPositions[self.currentPositionIndex % 4];
```

File: Examples/ObjectiveC/OrnamentsLayoutExample.m [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    // Update ornaments position with position matrix.
+    self.mapView.scaleBarPosition = currentPosition[0];
+    self.mapView.compassViewPosition = currentPosition[1];
+    self.mapView.logoViewPosition = currentPosition[2];
+    self.mapView.attributionButtonPosition = currentPosition[3];
+}
+
+- (void)onTimerTick {
+    self.currentPositionIndex ++;
```
Comment (friedbunny) [inline Examples/ObjectiveC/OrnamentsLayoutExample.m]:
```suggestion
    self.currentPositionIndex++;
```

**Comment by jmkiley** [comment]
Having the ornaments in set locations sounds good to me. Makes it more straightforward.

**Pull request closed without merging**